lib/http_perform.c

```diff
@@ -350,16 +347,29 @@ sf_bool STDCALL http_perform(CURL *curl,
                                     "Unable to get http response code",
                                     SF_SQLSTATE_UNABLE_TO_CONNECT);
             } else if (http_code != 200) {
-                retry = is_retryable_http_code(http_code);
-                if (!retry) {
-                    char msg[1024];
-                    sb_sprintf(msg, sizeof(msg), "Received unretryable http code: [%d]",
-                               http_code);
-                    SET_SNOWFLAKE_ERROR(error,
-                                        SF_STATUS_ERROR_RETRY,
-                                        msg,
-                                        SF_SQLSTATE_UNABLE_TO_CONNECT);
-                }
+              retry = is_retryable_http_code(http_code);
+              if (!retry) {
+                char msg[1024];
+                sb_sprintf(msg, sizeof(msg), "Received unretryable http code: [%d]",
+                           http_code);
+                SET_SNOWFLAKE_ERROR(error,
+                                    SF_STATUS_ERROR_RETRY,
+                                    msg,
+                                    SF_SQLSTATE_UNABLE_TO_CONNECT);
+              }
+              if (retry && curl_retry_ctx.retry_count < MAX_RETRY_COUNT)
+              {
+                uint32 next_sleep_in_secs = retry_ctx_next_sleep(&curl_retry_ctx);
+                log_debug(
+                    "curl_easy_perform() Got retryable error http code %d, retry count  %d "
+                    "will retry after %d micro seconds", http_code,
+                    curl_retry_ctx.retry_count,
+                    next_sleep_in_secs * 100);
+                my_sleep_ms(next_sleep_in_secs * 100);
```

Comment:
if `next_sleep_in_secs` is 1 then the process will sleep for 
`usleep( 1 * 100 * 1000)` instead of `usleep( 1 * 1000 * 1000)`